web-api/src/business/useCases/practitioner/getPractitionersByNameInteractor.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  ROLE_PERMISSIONS,
+  isAuthorized,
+} from '@shared/authorization/authorizationClientService';
+import { UnauthorizedError } from '@web-api/errors/errors';
+
+export type PractitionersByName = {
+  searchResults: {
+    lastKey: (string | number)[];
+    practitioners: {
+      admissionsDate: string;
+      admissionsStatus: string;
+      barNumber: string;
+      contact: {
+        state?: string;
+      };
+      name: string;
+      practiceType: string;
+      practitionerType: string;
+    }[];
+    total: number;
+  };
+};
+
+export const getPractitionersByNameInteractor = async (
+  applicationContext: IApplicationContext,
```

Comment:
Thank you for moving this interactor to web-api!

Nit: This can be ServerApplicationContext